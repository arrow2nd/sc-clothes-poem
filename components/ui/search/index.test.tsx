import { fireEvent, render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'

import Search from './index'

describe('Search', () => {
  const sleep = () => new Promise((r) => setTimeout(r, 50))

  const keyDownEnter = {
    key: 'Enter',
    code: 'Enter',
    charCode: 13
  }

  test('見た目が変化していないか', () => {
    const { container } = render(<Search onSearch={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  test('検索欄入力後に他の検索欄がクリアされるか', async () => {
    const mock = jest.fn()
    const { container, getByRole, getAllByRole } = render(
      <Search onSearch={mock} />
    )

    const textbox = getByRole('textbox')

    // アイドル名のコンボボックスを選択
    await act(async () => {
      const idolCombobox = getAllByRole('combobox')[0]

      fireEvent.change(idolCombobox, {
        target: { value: 'あさひ' }
      })

      await sleep()

      fireEvent.keyDown(idolCombobox, keyDownEnter)
    })

    expect(mock).toBeCalledTimes(1)
    expect(textbox).not.toHaveValue('もぎたて')
    expect(container).toHaveTextContent('芹沢あさひ')
    expect(container).not.toHaveTextContent('ジャージ')

    // 衣装名のコンボボックスを選択
    await act(async () => {
      const clothesCombobox = getAllByRole('combobox')[1]

      fireEvent.change(clothesCombobox, {
        target: { value: 'ジャージ' }
      })

      await sleep()

      fireEvent.keyDown(clothesCombobox, keyDownEnter)
    })

    expect(mock).toBeCalledTimes(2)
    expect(textbox).not.toHaveValue('もぎたて')
    expect(container).not.toHaveTextContent('芹沢あさひ')
    expect(container).toHaveTextContent('ジャージ')

    // 本文検索欄に入力して確定
    act(() => {
      fireEvent.change(textbox, {
        target: { value: 'もぎたて' }
      })

      fireEvent.keyDown(textbox, keyDownEnter)
    })

    expect(mock).toBeCalledTimes(3)
    expect(textbox).toHaveValue('もぎたて')
    expect(container).not.toHaveTextContent('芹沢あさひ')
    expect(container).not.toHaveTextContent('ジャージ')
  })
})
