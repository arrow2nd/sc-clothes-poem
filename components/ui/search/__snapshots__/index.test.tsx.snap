// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search 見た目が変化していないか 1`] = `
<div>
  <div
    class="flex justify-center mb-16"
  >
    <div
      class="w-full lg:w-auto max-w-lg lg:max-w-none"
    >
      <div
        class="flex flex-row items-center ml-2 mb-4 text-2xl text-main"
      >
        <svg
          class="mr-2"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        <span
          class="tracking-widest"
        >
          Search
        </span>
      </div>
      <div
        class="flex flex-wrap"
      >
        <div
          class="w-full lg:w-72 m-2"
        >
          <div
            class="flex items-center rounded-md border-2 border-main text-main bg-white"
          >
            <input
              class="w-full h-9 ml-2.5 placeholder-sub rounded-md focus:outline-none"
              placeholder="ポエムの一部から"
              type="text"
            />
            <button
              class="mx-1 p-2 transition-colors focus:outline-none"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <line
                  x1="21"
                  x2="16.65"
                  y1="21"
                  y2="16.65"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="w-full lg:w-72 m-2 bg-white css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-アイドル名から-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1ycg57x-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-1yaegve-placeholder"
                id="react-select-アイドル名から-placeholder"
              >
                <p>
                  アイドル名から
                </p>
              </div>
              <div
                class=" css-17k8k4u-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-アイドル名から-listbox"
                  aria-describedby="react-select-アイドル名から-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-owns="react-select-アイドル名から-listbox"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-アイドル名から-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-p70wra-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-lp1szy-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full lg:w-72 m-2 bg-white css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-衣装名から-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1ycg57x-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-1yaegve-placeholder"
                id="react-select-衣装名から-placeholder"
              >
                <p>
                  衣装名から
                </p>
              </div>
              <div
                class=" css-17k8k4u-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-衣装名から-listbox"
                  aria-describedby="react-select-衣装名から-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-owns="react-select-衣装名から-listbox"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-衣装名から-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-p70wra-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-lp1szy-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
